### ISSUE TITLE
Strategy analysis took 178.11s, more than 25% of the timeframe (75.00s)

### ISSUE THREAD

**Issue by riven-blade** [iterativv/NostalgiaForInfinity#412]
WARNING - Strategy analysis took 178.11s, more than 25% of the timeframe (75.00s). This can lead to delayed orders and missed signals.Consider either reducing the amount of work your strategy performs or reduce the amount of pairs in the Pairlist.

**Comment by DominiqueGEORGES** [comment]
Hi.

I also have this error message, but it does not have any impact on the way NFI performs

**Comment by riven-blade** [comment]
Hi. 
Are you sure ?
This will have a great impact on the ultra-short term. The NFI trading cycle is relatively long, but I am not sure whether it will affect the ultra-short term.

**Comment by stash86** [comment]
Imagine you have a green long candles. By the time you entered, you might be entering near the top of that green candle. Decide yourself whether buying near the top is good or not.

**Comment by iterativv** [comment]
> Hi. Are you sure ? This will have a great impact on the ultra-short term. The NFI trading cycle is relatively long, but I am not sure whether it will affect the ultra-short term.

Hi, in long term it doesn't matter. Some times it's better to delay some times is better to be faster.

There will be a new version, that for this part at least, is going to use all cores. Live ready in 2-3 weeks now.

**Comment by roko772** [comment]
> There will be a new version, that for this part at least, is going to use all cores. Live ready in 2-3 weeks now.

Nobody asked, everybody needed - that's all I can say. Thank you!

**Comment by roko772** [comment]
Perhaps an OffsetFilter can be used to divide pairlist between multiple instances.

**Comment by iterativv** [comment]
> Perhaps an OffsetFilter can be used to divide pairlist between multiple instances.

Yes, that been done. Anyway, there will be no issues with CPU usage for X5. In 2-3 weeks should be ready for the first live runs.

**Comment by veyakup** [comment]
> > Perhaps an OffsetFilter can be used to divide pairlist between multiple instances.
> 
> Yes, that been done. Anyway, there will be no issues with CPU usage for X5. In 2-3 weeks should be ready for the first live runs.

First of all, thank you very much for your effort. Have you updated? I'm still getting this error.

**Comment by iterativv** [comment]
Hi, X6 should be a lot faster now. Thank you !
